### PR TITLE
Document substring, position, upper, and lower functions

### PR DESCRIPTION
**Pull request by ccobb-nr**
This PR adds documentation of the following new NRQL functions:
* lower()
* position()
* substring()
* upper()

Please follow [conventional commit standards](https://www.conventionalcommits.org/en/v1.0.0/)
in your commit messages and pull request title.

## Give us some context

* What problems does this PR solve?
* Add any context that will help us review your changes such as testing notes,
  links to related docs, screenshots, etc.
* If your issue relates to an existing GitHub issue, please link to it.

### CODE REVIEW

**Comment by ccobb-nr** [comment]
FYI, we are postponing the GA of the NRQL methods added in this PR, so this will likely stay as a Draft for at least several weeks.

File: src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions.mdx [hunk 1]
```diff
@@ -2426,10 +2434,98 @@ Note: `aparse()` is case-insensitive.
     </CollapserGroup>
   </Collapser>
 
+  <Collapser
+    className="freq-link"
+    id="func-lower"
+    title={<InlineCode>lower(str)</InlineCode>}
+  >
+    Use the lower() function transliterates all alphabetic characters of a string value to lower-case.
```
Comment (mlobiondo-newrelic) [inline src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions.mdx:2442]: _to transliterate_

Comment (mlobiondo-newrelic) [reply src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions.mdx:2442]: "Transliterate" is kind of fancy, maybe use "change"?

File: src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions.mdx [hunk 2]
```diff
@@ -2426,10 +2434,98 @@ Note: `aparse()` is case-insensitive.
...
+
+    Arguments:
+    * str - The string value to be transliterated to lower-case
+      * This can be anything that evaluates to a string, including a literal string in quotes, a queried string attribute, a function that returns a string, or even a subquery that returns a single string value.
+      * If this argument evaluates to null, the lower function will return null.
```
Comment (mlobiondo-newrelic) [inline src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions.mdx:2447]: All instances of function names like "lower" should have parens and backticks, for consistency with other entries (e.g. `aggregationendtime()`)

File: src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions.mdx [hunk 3]
```diff
@@ -2426,10 +2434,98 @@ Note: `aparse()` is case-insensitive.
...
+
+    <CollapserGroup>
+      <Collapser title="Use lower to transliterate a string value to all lower-case characters">
+        This query demonstrates use of the lower function in various parts of a query.
+
+        ```
```
Comment (mlobiondo-newrelic) [inline src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions.mdx:2453]: "```sql" markup is used for queries

File: src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions.mdx [hunk 4]
```diff
@@ -2458,11 +2554,94 @@ Note: `aparse()` is case-insensitive.
...
+      * The first character in str is position 0.
+      * A negative value will find the position relative to the end of str, with the last character of the string being position -1.
+      * If start is larger or equal to the length of str, the substring function will return an empty string.
+      * If start is negative, and its absolute value is larger than the length of str, the extracted substring will begin at position 0.
+    * length - the length, or number of characters, of the substring to extract from str.
+      * _Optional_ - if length is not provided, all characters in str after the resolved start position will be included.
+
+    <CollapserGroup>
+      <Collapser title="Use substring to extract various parts of a string">
+        This query returns various parts of session value.
```
Comment (mlobiondo-newrelic) [inline src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions.mdx:2584]: of _the_ session value

Comment (mlobiondo-newrelic) [reply src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions.mdx:2584]: I think you could remove "various" from both lines without affecting the meaning.

File: src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions.mdx [hunk 5]
```diff
@@ -2458,11 +2554,94 @@ Note: `aparse()` is case-insensitive.
...
     </CollapserGroup>
+    _See the [<InlineCode>position</InlineCode>](#func-position) function for examples of using <InlineCode>substring</InlineCode> and <InlineCode>position</InlineCode> together._
   </Collapser>
+
+  <Collapser
+    className="freq-link"
+    id="func-upper"
+    title={<InlineCode>upper(str)</InlineCode>}
+  >
+    Use the upper() function transliterates all alphabetic characters of a string value to upper-case.
```
Comment (mlobiondo-newrelic) [inline src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions.mdx:2612]: _to transliterate_ (or "to change", as above)

File: src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions.mdx (hunk 1 above)
Comment (mlobiondo-newrelic) [reply src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions.mdx:2442]: I believe "lower case" would be correct here; "lower-case" is for the verb or adjective form.

File: src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions.mdx (hunk 5 above)
Comment (mlobiondo-newrelic) [reply src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions.mdx:2612]: Also same re: "upper-case" -> "upper case"

**Review by mlobiondo-newrelic — APPROVED** [review]

**Review by mlobiondo-newrelic — APPROVED** [review]

**Review by akristen — APPROVED** [review]

**Pull request merged**